src/generic/UnivPoly.jl

```diff
@@ -244,17 +244,29 @@ total_degree(p::UnivPoly) = total_degree(data(p))
 
 length(p::UnivPoly) = length(data(p))
 
-function gen(S::UniversalPolyRing{T}, s::VarName) where {T <: RingElement}
-   i = findfirst(==(Symbol(s)), S.S)
-   if i === nothing
-      push!(S.S, Symbol(s))
+function _ensure_variables(S::UniversalPolyRing, v::Vector{<:VarName})
+   idx = Int[]
+   for s in v
+    i = findfirst(==(Symbol(s)), S.S)
+    if i === nothing
+       push!(S.S, Symbol(s))
```

Comment:
```suggestion
   for s_ in v
    s = Symbol(s_)
    i = findfirst(==(s), S.S)
    if i === nothing
       push!(S.S, s)
```
one minor nitpick: you convert `s` twice into a Symbol. This can be avoided with this patch

Reply:
Thanks!